pkg/virt-controller/ipamclaims/ipamclaims_test.go

```diff
@@ -0,0 +1,305 @@
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			ownerRef := &v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachineInstance",
+				Name:               vmi.Name,
+				UID:                vmiUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}
+			Expect(ipamClaimsManager.CreateIPAMClaims(vmi.Namespace, vmi.Name, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, ownerRef)).To(Succeed())
+
+			ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(redNamespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(2))
+			assertIPAMClaim(ipamClaimsList.Items[0], vmi.Name, blueNetworkLogicalName, "pod16477688c0e")
+			assertIPAMClaim(ipamClaimsList.Items[1], vmi.Name, redNetworkLogicalName, "podb1f51a511f1")
+		})
+
+		Context("When IPAMClaims already exist", func() {
+			var ownerRef *v1.OwnerReference
+
+			BeforeEach(func() {
+				ownerRef = &v1.OwnerReference{
+					APIVersion:         "kubevirt.io/v1",
+					Kind:               "VirtualMachineInstance",
+					Name:               vmi.Name,
+					UID:                vmiUID,
+					Controller:         pointer.P(true),
+					BlockOwnerDeletion: pointer.P(true),
+				}
+				Expect(ipamClaimsManager.CreateIPAMClaims(vmi.Namespace,
+					vmi.Name,
+					vmi.Spec.Domain.Devices.Interfaces,
+					vmi.Spec.Networks,
+					ownerRef)).To(Succeed())
+			})
+
+			It("with the expected owner UID, should not fail re-creation/validation attempt", func() {
+				Expect(ipamClaimsManager.CreateIPAMClaims(
+					vmi.Namespace,
+					vmi.Name,
+					vmi.Spec.Domain.Devices.Interfaces,
+					vmi.Spec.Networks,
+					ownerRef)).To(Succeed())
+			})
+
+			It("with a different owner UID, should fail fail re-creation/validation attempt", func() {
+				ownerRef.UID = "differentUID"
+				err := ipamClaimsManager.CreateIPAMClaims(vmi.Namespace,
+					vmi.Name,
+					vmi.Spec.Domain.Devices.Interfaces,
+					vmi.Spec.Networks,
+					ownerRef)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("wrong IPAMClaim with the same name still exists"))
+			})
+		})
+	})
+
+	Context("With allowPersistentIPs disabled in the NADs", func() {
+		BeforeEach(func() {
+			Expect(createNADs(networkClient, redNamespace, vmi.Spec.Networks, map[string]struct{}{})).To(Succeed())
+		})
+
+		It("should not create IPAMClaims", func() {
+			ownerRef := &v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachineInstance",
+				Name:               vmi.Name,
+				UID:                vmiUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}
+			ipamClaimsManager := ipamclaims.NewIPAMClaimsManager(networkClient, fakeipamclaimclient.NewSimpleClientset())
```

Comment:
use `ipamClaimsClient` instead of creating a new fake ipam claims client.

Reply:
right leftover done